File: Java/src/main/java/br/ufrn/imd/config/ExecutionConfig.java
package br.ufrn.imd.config;

import br.ufrn.imd.config.enums.SyncType;
import br.ufrn.imd.config.enums.ThreadType;

/**
 * Configuração imutável de um cenário de execução IDW. Viaja por todo o
 * pipeline (leitura do CSV, particionamento, fase de cálculo).
 *
 * @param name              rótulo legível do cenário (usado em logs/relatórios)
 * @param ioThreadType      tipo de thread usado pelo {@code CsvDatasetReader}
 * @param ioThreads         quantidade de threads para parsear o CSV em paralelo
 * @param computeThreadType tipo de thread usado pelo executor da fase de cálculo
 * @param computeThreads    quantidade de threads/workers da fase de cálculo
 * @param syncType          estratégia de sincronização dos acumuladores
 * @param chunkSize         tamanho do chunk de particionamento (sensores ou ranges PC)
 */
public record ExecutionConfig(
        String name,

        ThreadType ioThreadType,
        int ioThreads,

        ThreadType computeThreadType,
        int computeThreads,

        SyncType syncType,

        int chunkSize
) { }

File: Ruby/lib/idw/execution_config.rb
# frozen_string_literal: true

module IDW
  # Configuração imutável de um cenário — espelha Java/.../config/ExecutionConfig.java.
  #
  # @!attribute [r] name
  # @!attribute [r] io_thread_type    :platform | :virtual
  # @!attribute [r] io_threads
  # @!attribute [r] compute_thread_type :platform | :virtual
  # @!attribute [r] compute_threads
  # @!attribute [r] sync_type         :none | :mutex | :semaphore | :volatile | :atomic | :producer_consumer
  # @!attribute [r] chunk_size
  ExecutionConfig = Data.define(
    :name,
    :io_thread_type,
    :io_threads,
    :compute_thread_type,
    :compute_threads,
    :sync_type,
    :chunk_size
  )
end
